core/src/main/java/org/apache/iceberg/RESTFileScanTaskParser.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.ExpressionParser;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.Sets;
+import org.apache.iceberg.util.JsonUtil;
+
+public class RESTFileScanTaskParser {
```

Comment:
On second look, shouldn't this be in the REST package?

Reply:
If we move this to the REST package then we run into issues that we saw in the past with the Data and Delete Files being package private.

<img width="425" alt="Screenshot 2024-11-27 at 11 53 09 AM" src="https://github.com/user-attachments/assets/737ea1eb-a60b-4808-be62-c00b5031d3e1">

I think we can keep where it is currently, so we can avoid making anything public.

Reply:
Thanks for the reminder, I think in my PR to your branch, I forgot to address this part I think it's worth seeing how we can refactor this so that we can actually move `RESTFileScanTaskParser` in the REST package...it kind of stands out by not really following existing convention